### ISSUE TITLE
Model signature tests fail on zOS

### ISSUE THREAD

**Issue by JonTriebenbach** [onnx/onnx-mlir#1310]
The model signature tests hit this error on zOS:
```
JLT@PLPSC:dlc-automation$ java -classpath $HOME/dlc-automation/client/class:$HOME/dlc-automation/models/CPU/mnist-8.jar input_output
[2022-04-01 15:39:58 -0400][2460600000000003][error]jniwrapper.c:Java_com_ibm_onnxmlir_OMModel_input_1signature_1jni:794 jni_isig=0
null|[2022-04-01 15:39:58 -0400][2460600000000003][error]jniwrapper.c:Java_com_ibm_onnxmlir_OMModel_output_1signature_1jni:836 jni_osig=0
```
Doesn't show much except that both the input and output signature calls fail.  Only the jar model tests fail.  The so shared object tests pass.

The tests are run with the modelzoo mnist-8 model.

The tests are calling the src/Runtime/jni/src/com/ibm/onnxmlir/OMModel.java API's:
```
    public static String inputSignature() {
   	return input_signature_jni("run_main_graph");

    public static String outputSignature() {
  	return output_signature_jni("run_main_graph");
```
The problem is that src/Runtime/jni/jniwrapper.c does not convert the entry point name to EBCDIC prior to calling omInputSignature() or omOutputSignature().

I added the code below to jniwrapper.c which corrected the issue:
```
  return dupsig;
}

static inline char *ep_a2e(const char *origep) {
  char *dupep =strdup(origep);
  if (dupep == NULL)
    return NULL;

  /* Convert ASCII to EBCDIC in place */
  if (__a2e_s(dupep) < 0) {
    free(dupep);
    return NULL;
  }

  return dupep;
}
#endif
```
```
#ifdef __MVS__
  CHECK_CALL(char *, ep_ptr, ep_a2e(jni_ep), ep_ptr != NULL, "ep_ptr=%p", ep_ptr);
#else
  const char *ep_ptr = jni_ep;
#endif
  /* Call model input signature API */
  CHECK_CALL(const char *, jni_isig, omInputSignature(ep_ptr), jni_isig != NULL,
      "jni_isig=%p", jni_isig);
```
```
#ifdef __MVS__
  CHECK_CALL(char *, ep_ptr, ep_a2e(jni_ep), ep_ptr != NULL, "ep_ptr=%p", ep_ptr);
#else
  const char *ep_ptr = jni_ep;
#endif
  /* Call model output signature API */
  CHECK_CALL(const char *, jni_osig, omOutputSignature(ep_ptr),
      jni_osig != NULL, "jni_osig=%p", jni_osig);
```
The output with the corrected code is below:
```
JLT@PLPSC:dlc-automation$ java -classpath $HOME/dlc-automation/client/class:$HOME/dlc-automation/models/CPU/mnist-8.jar input_output
[    { "type" : "f32" , "dims" : [1 , 1 , 28 , 28] , "name" : "Input3" }

]|[   { "type" : "f32" , "dims" : [1 , 10] , "name" : "Plus214_Output_0" }
```
The reason the entry point needs to be converted into EBCDIC follows from similar code in jniwrapper.c on the:
```
Java_com_ibm_onnxmlir_OMModel_query_1entry_1points(JNIEnv *env, jclass cls) {
```
```
  /* Go through the entry points, convert them into Java String and
   * put them into the String array.
   */
  for (int64_t i = 0; i < neps; i++) {
    char ep[32];
    sprintf(ep, "ep[%lld]", i);
    HEX_DEBUG(ep, jni_eps[i], strlen(jni_eps[i]));
    LOG_PRINTF(LOG_DEBUG, "ep[%d](%ld):%s", i, strlen(jni_eps[i]), jni_eps[i]);

#ifdef __MVS__
    CHECK_CALL(
	char *, epptr, sig_e2a(jni_eps[i]), epptr != NULL, "epptr=%p", epptr);
#else
    const char *epptr = jni_eps[i];
#endif
```
The entry points are converted from EBCDIC to UTF-8 to return the entry points to Java.  When passing the entry point names into the model, the entry point names then need to be first converted back into EBCDIC from UTF8.

**Comment by gongsu832** [comment]
As I mentioned in the slack channel. You are mixing things up. On z/OS, if you have passed in the correct mtriple indicating z/OS target when cross compiling, `omInputSignature` and `omOutputSignature` already return EBCDIC. There is no need to convert.

**Comment by gongsu832** [comment]
> The problem is that src/Runtime/jni/jniwrapper.c does not convert the entry point name to EBCDIC prior to calling omInputSignature() or omOutputSignature().

Are you compiling `src/Runtime/jni/jniwrapper.c` on zLinux? I thought you would compile it on z/OS, in which case all string literals would be encoded in EBCDIC (unless of course you override it).

**Comment by gongsu832** [comment]
Should be fixed by #1313
